mmocr/models/textrecog/layers/res_blocks.py

```diff
@@ -0,0 +1,134 @@
+    return nn.Conv2d(
+        in_planes, out_planes, kernel_size=1, stride=1, padding=0, bias=False)
+
+
+class BasicBlock_New(nn.Module):
+
+    def __init__(
+        self,
+        cfg,
+        inplanes,
+        planes,
+        stride=1,
+        downsample=None,
+    ):
+        super(BasicBlock_New, self).__init__()
+
+        if 'use_conv1x1' in cfg and cfg['use_conv1x1'] is True:
+            self.conv1 = conv1x1(inplanes, planes)
+            self.conv2 = conv3x3(planes, planes, stride)
+        else:
+            self.conv1 = conv3x3(inplanes, planes, stride)
+            self.conv2 = conv3x3(planes, planes)
+
+        self.plugin = None
+        if 'plugins' in cfg and cfg['plugins'] is not None:
+            self.plugin = self._make_block_plugins(cfg['plugins'])
+        self.planes = planes
+        self.bn1 = nn.BatchNorm2d(planes)
+        self.relu = nn.ReLU(inplace=True)
+        self.bn2 = nn.BatchNorm2d(planes)
+        self.downsample = downsample
+        self.stride = stride
+
+    def _creat_plugin(self, plugin_cfg):
+        if plugin_cfg['type'] == 'Maxpooling':
+            return nn.MaxPool2d(plugin_cfg['arg'])
+        elif plugin_cfg['type'] == 'Conv':
+            return nn.Sequential(
+                nn.Conv2d(
+                    self.planes, self.planes, **plugin_cfg['arg'], bias=False),
+                nn.BatchNorm2d(self.planes), nn.ReLU(inplace=True))
+        elif plugin_cfg['type'] == 'GCA':
+            raise Exception('{} not implemented yet'.format(plugin_cfg['arg']))
+
+    def _make_block_plugins(self, plugins):
+        """Make plugins for ResNet ``stage_idx`` th stage.
+        Currently we support to insert ``nn.Maxpooling``,
+        ``nn.conv2d``into the backbone. This is designed for ResNet31
+        architecture likes
+        An example of plugins format could be:
+        Examples:
+            >>> plugins=[
+            ...     dict(cfg=dict(type="Maxpooling", stride=(2,2)),
+            ...          position='before_conv1'),
+            ...     dict(cfg=dict(type="Maxpooling", stride=(2,1)),
+            ...          position='before_conv2'),
+            ...     dict(cfg=dict(type="Conv",
+            ...          args=(dict(kernel_size=3, stride=1, padding=1))),
+            ...          position=('after_conv2')
+            ... ]
+        Suppose ``stage_idx=1``, the structure of stage would be:
+        .. code-block:: none
+            Maxpooling-> Basicbloks * blocks[i] -> Conv
+        Suppose 'stage_idx=1', the structure of blocks in the stage would be:
+        If stages is missing, the plugin would be applied to all stages.
+        Args:
+            plugins (list[dict]): List of plugins cfg to build. The postfix is
+                required if multiple same type plugins are inserted.
+            stage_idx (int): Index of stage to build
+        Returns:
+            list[dict]: Plugins for current stage
+        """
+
+        exist_position = []
```

Comment:
```suggestion
        allowed_position = [...]
```

Reply:
Well, I think we can use occupied_postition instead?  I use exist_position to record which position has already been occupied and will not be allowed to insert new plugins.